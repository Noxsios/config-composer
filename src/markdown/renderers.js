import { EuiCodeBlock, EuiCode, EuiText, EuiTextColor, EuiHorizontalRule } from "@elastic/eui";
const renderers = {
  root: ({ children }) => {
    return <EuiText>{children}</EuiText>;
  },
  paragraph: ({ children }) => {
    return <EuiText>{children}</EuiText>;
  },
  strong: ({ children }) => {
    return (
      <EuiTextColor color="secondary">
        <strong>{children}</strong>
      </EuiTextColor>
    );
  },
  blockquote: ({ children }) => {
    return (
      <EuiText style={{ margin: "0.5rem" }}>
        <blockquote>{children}</blockquote>
      </EuiText>
    );
  },
  emphasis: ({ children }) => {
    return (
      <EuiTextColor color="accent">
        <em>{children}</em>
      </EuiTextColor>
    );
  },
  code: ({ language, value }) => {
    return <EuiCodeBlock language={language} isCopyable paddingSize="m" children={value} />;
  },
  thematicBreak: () => {
    return <EuiHorizontalRule size="half" />;
  },
  inlineCode: ({ language = "js", value }) => {
    return <EuiCode language={language} children={value} style={{ color: "#FFCE7A" }} />;
  },
};

export { renderers };
