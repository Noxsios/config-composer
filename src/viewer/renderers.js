import {
  EuiCodeBlock,
  EuiCode,
  EuiText,
  EuiTextColor,
  EuiHorizontalRule,
  EuiTableHeader,
  EuiTableBody,
  EuiTableRow,
  EuiTableRowCell,
} from "@elastic/eui";
const renderers = {
  root: ({ children }) => {
    return <EuiText>{children}</EuiText>;
  },
  paragraph: ({ children }) => {
    return <EuiText style={{ marginBottom: "0.5rem" }}>{children}</EuiText>;
  },
  strong: ({ children }) => {
    return (
      <EuiTextColor color="secondary">
        <strong>{children}</strong>
      </EuiTextColor>
    );
  },
  blockquote: ({ children }) => {
    return (
      <EuiText style={{ margin: "0.5rem" }}>
        <blockquote>{children}</blockquote>
      </EuiText>
    );
  },
  emphasis: ({ children }) => {
    return (
      <EuiTextColor color="accent">
        <em>{children}</em>
      </EuiTextColor>
    );
  },
  code: ({ language, value }) => {
    return (
      <div style={{ marginLeft: "2rem", marginRight: "2rem" }}>
        <EuiCodeBlock
          language={language}
          isCopyable
          paddingSize="m"
          children={value}
        />
      </div>
    );
  },
  thematicBreak: () => {
    return <EuiHorizontalRule size="half" />;
  },
  inlineCode: ({ language = "js", value }) => {
    return (
      <EuiCode
        language={language}
        children={value}
        style={{ color: "#FFCE7A" }}
      />
    );
  },
  table: ({ children }) => {
    return (
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
        }}
      >
        <table style={{ display: "initial" }}>{children}</table>
      </div>
    );
  },
  tableHead: ({ children }) => {
    return (
      <EuiTableHeader style={{ fontWeight: "bold" }}>{children}</EuiTableHeader>
    );
  },
  tableBody: ({ children }) => {
    return <EuiTableBody>{children}</EuiTableBody>;
  },
  tableRow: ({ children }) => {
    return <EuiTableRow>{children}</EuiTableRow>;
  },
  tableCell: ({ children }) => {
    return <EuiTableRowCell>{children}</EuiTableRowCell>;
  },
};

export { renderers };
