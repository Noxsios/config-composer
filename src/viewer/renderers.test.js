import renderer from "react-test-renderer";
import { renderers } from "./renderers";

const coffeeIpsum =
  "Single shot and grounds, java aged rich spoon arabica flavour. Percolator eu, strong lungo ristretto saucer instant beans. Robust seasonal brewed mocha, extraction barista milk as mocha decaffeinated latte.";

describe("Markdown", () => {
  it("renders root", () => {
    const t = renderer
      .create(<renderers.root children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders p", () => {
    const t = renderer
      .create(<renderers.paragraph children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders strong", () => {
    const t = renderer
      .create(<renderers.strong children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders blockquote", () => {
    const t = renderer
      .create(<renderers.blockquote children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders em", () => {
    const t = renderer
      .create(<renderers.emphasis children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders codeblock", () => {
    const t = renderer
      .create(<renderers.code children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders hr", () => {
    const t = renderer.create(<renderers.thematicBreak />).toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders inlineCode", () => {
    const t = renderer
      .create(<renderers.inlineCode children={coffeeIpsum} />)
      .toJSON();
    expect(t).toMatchSnapshot();
  });
  it("renders table", () => {
    const SampleTable = () => {
      return (
        <renderers.table>
          <renderers.tableHead>
            <renderers.tableRow>
              <renderers.tableCell>Title</renderers.tableCell>
            </renderers.tableRow>
          </renderers.tableHead>
          <renderers.tableBody>
            <renderers.tableRow>
              <renderers.tableCell>Body</renderers.tableCell>
            </renderers.tableRow>
          </renderers.tableBody>
        </renderers.table>
      );
    };
    const t = renderer.create(<SampleTable />).toJSON();
    expect(t).toMatchSnapshot();
  });
});
