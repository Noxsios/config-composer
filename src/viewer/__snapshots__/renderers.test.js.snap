// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renders blockquote 1`] = `
<div
  className="euiText euiText--medium"
  style={
    Object {
      "margin": "0.5rem",
    }
  }
>
  <blockquote>
    Single shot and grounds, java aged rich spoon arabica flavour. Percolator eu, strong lungo ristretto saucer instant beans. Robust seasonal brewed mocha, extraction barista milk as mocha decaffeinated latte.
  </blockquote>
</div>
`;

exports[`Markdown renders codeblock 1`] = `
<div
  style={
    Object {
      "marginLeft": "2rem",
      "marginRight": "2rem",
    }
  }
/>
`;

exports[`Markdown renders em 1`] = `
<span
  className="euiTextColor euiTextColor--accent"
>
  <em>
    Single shot and grounds, java aged rich spoon arabica flavour. Percolator eu, strong lungo ristretto saucer instant beans. Robust seasonal brewed mocha, extraction barista milk as mocha decaffeinated latte.
  </em>
</span>
`;

exports[`Markdown renders hr 1`] = `
<hr
  className="euiHorizontalRule euiHorizontalRule--half euiHorizontalRule--marginLarge"
/>
`;

exports[`Markdown renders inlineCode 1`] = `null`;

exports[`Markdown renders p 1`] = `
<div
  className="euiText euiText--medium"
  style={
    Object {
      "marginBottom": "0.5rem",
    }
  }
>
  Single shot and grounds, java aged rich spoon arabica flavour. Percolator eu, strong lungo ristretto saucer instant beans. Robust seasonal brewed mocha, extraction barista milk as mocha decaffeinated latte.
</div>
`;

exports[`Markdown renders root 1`] = `
<div
  className="euiText euiText--medium"
>
  Single shot and grounds, java aged rich spoon arabica flavour. Percolator eu, strong lungo ristretto saucer instant beans. Robust seasonal brewed mocha, extraction barista milk as mocha decaffeinated latte.
</div>
`;

exports[`Markdown renders strong 1`] = `
<span
  className="euiTextColor euiTextColor--secondary"
>
  <strong>
    Single shot and grounds, java aged rich spoon arabica flavour. Percolator eu, strong lungo ristretto saucer instant beans. Robust seasonal brewed mocha, extraction barista milk as mocha decaffeinated latte.
  </strong>
</span>
`;

exports[`Markdown renders table 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <table
    style={
      Object {
        "display": "initial",
      }
    }
  >
    <thead
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      <tr>
        <tr
          className="euiTableRow"
        >
          <td
            className="euiTableRowCell"
            style={
              Object {
                "width": undefined,
              }
            }
          >
            <div
              className="euiTableCellContent"
            >
              <span
                className="euiTableCellContent__text"
              >
                Title
              </span>
            </div>
          </td>
        </tr>
      </tr>
    </thead>
    <tbody>
      <tr
        className="euiTableRow"
      >
        <td
          className="euiTableRowCell"
          style={
            Object {
              "width": undefined,
            }
          }
        >
          <div
            className="euiTableCellContent"
          >
            <span
              className="euiTableCellContent__text"
            >
              Body
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
